DuckDuckGo/FeedbackViewController.swift

```diff
@@ -0,0 +1,179 @@
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import Foundation
+
+import UIKit
+import Core
+import ToastSwiftFramework
+
+class FeedbackViewController: UIViewController {
+    
+    private struct ViewConstants {
+        static let urlTextHeight: CGFloat = 38
+        static let urlTextPadding: CGFloat = 4
+    }
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var brokenSiteSwitch: UISwitch!
+    @IBOutlet weak var urlTextFieldHeight: NSLayoutConstraint!
+    @IBOutlet weak var urlTextField: UITextField!
+    @IBOutlet weak var messageTextView: UITextView!
+    @IBOutlet weak var messagePlaceholderText: UILabel!
+    @IBOutlet weak var submitButton: UIButton!
+    
+    private var feedbackModel = FeedbackModel()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.blur(style: .dark)
+        loadModel()
+        configureViews()
+        registerForKeyboardNotifications()
+        refreshMode()
+    }
+    
+    func prepareForSegue(isBrokenSite: Bool, url: String?) {
+        feedbackModel.isBrokenSite = isBrokenSite
+        feedbackModel.url = url
+    }
+    
+    private func loadModel() {
+        brokenSiteSwitch.isOn = feedbackModel.isBrokenSite
+        messageTextView.text = feedbackModel.message
+        urlTextField.text = feedbackModel.url
+    }
+    
+    private func configureViews() {
+        messageTextView.delegate = self
+        messageTextView.layer.borderWidth = 1
+        messageTextView.layer.borderColor = UIColor.mercury.cgColor
+        urlTextField.layer.borderWidth = 1
+        urlTextField.layer.borderColor = UIColor.mercury.cgColor
+        urlTextField.layer.sublayerTransform = CATransform3DMakeTranslation(ViewConstants.urlTextPadding, 0, 0)
+    }
+    
+    private func registerForKeyboardNotifications() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardDidShow),
+                                               name: NSNotification.Name.UIKeyboardDidShow,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillHide),
+                                               name: NSNotification.Name.UIKeyboardWillHide,
+                                               object: nil)
+    }
+    
+    @objc private func keyboardDidShow(notification: NSNotification) {
```

Comment:
Also would it be possible to tap out of the fields to hide the keyboard? 

That feels a bit more intuitive than scrolling to me.

Reply:
Nice idea, done!